### PR TITLE
Documentation for the ExecutionStore

### PR DESCRIPTION
**Pull request by ritch**
Documentation for the ExecutionStore

## Summary by CodeRabbit

- **Documentation**
	- Enhanced guidance on developing plugins with a new section on "Execution Store" for data persistence.
	- Added subsections on "Saved Workspaces," "Accessing Secrets," and "Common Patterns" for improved panel management.
	- Refined document structure for clarity and usability, with updated examples reflecting best practices.
	- Minor formatting adjustments made throughout the document to enhance readability.

### CODE REVIEW

File: docs/source/plugins/developing_plugins.rst [hunk 1]
```diff
@@ -2430,6 +2430,76 @@ loaded only when the `brain_key` property is modified.
...
+user preferences. See the
+:class:`ExecutionStore <fiftyone.operators.store.ExecutionStore>` and
+:class:`ExecutionStoreService <fiftyone.operators.store.ExecutionStoreService>`
+classes for more information.
+
+.. code-block:: python
+    :linenos:
+
+    from bson import ObjectId
+    from fiftyone.operators.store import ExecutionStore, ExecutionStoreService
```
Comment (tom-vx51) [inline docs/source/plugins/developing_plugins.rst:2450]: [nit] `ExecutionStore` is unused

**Review by tom-vx51 — DISMISSED** [review]

**Review by Br2850 — APPROVED** [review]

File: fiftyone/operators/store/store.py [hunk 2]
```diff
@@ -101,19 +102,24 @@ def update_ttl(self, key: str, new_ttl: int) -> None:
         """
         self._store_service.update_ttl(self.store_name, key, new_ttl)
 
-    def get_ttl(self, key: str) -> Optional[int]:
```
Comment (brimoor) [inline fiftyone/operators/store/store.py:104]: Realized in testing that this method was broken

File: fiftyone/operators/__init__.py [hunk 3]
```diff
@@ -19,6 +19,7 @@
 )
 from .utils import ProgressHandler, is_new
 from .panel import Panel, PanelConfig
+from .store import ExecutionStore
```
Comment (brimoor) [inline fiftyone/operators/__init__.py:22]: By convention, I prefer that anything that we're publicly documenting for users to be accessible via the highest reasonable package level. For example in this case:

```py
# happy
from fiftyone.operators import ExecutionStore

# sad; too low-level for end users
from fiftyone.operators.store import ExecutionStore
```

File: fiftyone/factory/repos/execution_store.py [hunk 4]
```diff
@@ -233,9 +221,7 @@ def has_store_global(self, store_name):
         """Determines whether a store with the given name exists across all
         datasets and the global context.
         """
-        result = self._collection.find_one(
```
Comment (brimoor) [inline fiftyone/factory/repos/execution_store.py:236]: Realized in testing that this method was broken

File: fiftyone/factory/repos/execution_store.py [hunk 5]
```diff
@@ -78,33 +78,21 @@ def has_store(self, store_name) -> bool:
 
     def list_stores(self) -> list[str]:
         """Lists the stores associated with the current context."""
-        result = self._collection.find(
```
Comment (brimoor) [inline fiftyone/factory/repos/execution_store.py:81]: Realized in testing that this method was broken

**Review by brimoor — APPROVED** [review]
LGTM

**Review by brimoor — APPROVED** [review]

**Pull request merged**